### PR TITLE
feat: add support for BSC

### PR DESCRIPTION
**Pull request by Pythonberg1997**
### Description

This pr is to add support for BSC

### Rationale

tell us why we need these changes...

### Example

add an example CLI or API response...

### Changes

Notable changes: 
* add BSC spec ids
* add BSC evm handlers
* add BSC precompile contracts
* other BSC related changes

### Potential Impacts
* add potential impacts for other components here
* ...

### CODE REVIEW

File: crates/revm/src/context/evm_context.rs [hunk 1]
```diff
@@ -121,8 +122,14 @@ impl<DB: Database> EvmContext<DB> {
         match out {
             Ok((gas_used, data)) => {
                 if result.gas.record_cost(gas_used) {
-                    result.result = InstructionResult::Return;
-                    result.output = data;
+                    // to keep align with bsc, revert if data is empty.
+                    // revert will not cost all gas
+                    if is_bsc_precompile(address) && data.is_empty() {
```
Comment (forcodedancing) [inline crates/revm/src/context/evm_context.rs:127]: Maybe, we can write it with feature to make it more clear.

File: crates/precompile/Cargo.toml [hunk 2]
```diff
@@ -46,6 +46,13 @@ cometbft-proto = {  git = "https://github.com/bnb-chain/greenfield-cometbft-rs.g
 cometbft-light-client = {  git = "https://github.com/bnb-chain/greenfield-cometbft-rs.git", rev = "1282547" }
 prost = { version = "0.12.3" }
 bls_on_arkworks = "0.3.0"
+tendermint = { git = "https://github.com/forcodedancing/tendermint-rs-parlia", rev = "10cfd14bac4cd4f1180a6670ea44edabfeda2519", features = ["secp256k1"] }
```
Comment (unclezoro) [inline crates/precompile/Cargo.toml:49]: `forcodedancing`?  We need an official repo

File: crates/primitives/src/specification.rs [hunk 3]
```diff
@@ -71,6 +74,56 @@ pub enum SpecId {
...
+    BERLIN = 21,         // Berlin                               31302048
+    LONDON = 22,         // London                               31302048
+    HERTZ = 23,          // Hertz                                31302048
+    HERTZ_FIX = 24,      // HertzFix                             34140700
+    SHANGHAI = 25,       // Shanghai                             timestamp(1705996800)
+    KEPLER = 26,         // Kepler                               timestamp(1705996800)
+    FEYNMAN = 27,        // Feynman                              timestamp(1713419340)
+    FEYNMAN_FIX = 28,    // FeynmanFix                           timestamp(1713419340)
+    CANCUN = 29,         // Cancun                               timestamp(1718863500)
+    HABER = 30,          // Haber                                TBD
```
Comment (unclezoro) [inline crates/primitives/src/specification.rs:113]: I think we have the `haber` time on BSC mainnet now.

File: crates/precompile/Cargo.toml (hunk 2 above)
Comment (Pythonberg1997) [reply crates/precompile/Cargo.toml:49]: Fixed

**Review by unclezoro — APPROVED** [review]

**Pull request merged**